wallet/src/de/schildbach/wallet/ui/WalletActivity.java

```diff
@@ -543,6 +544,46 @@ public void onDismiss(DialogInterface dialog) {
                     REQUEST_CODE_BACKUP_WALLET);
     }
 
+    private void handleResetWallet() {
+        android.app.AlertDialog.Builder dialogBuilder = new android.app.AlertDialog.Builder(this);
+        dialogBuilder.setTitle(R.string.wallet_lock_reset_wallet_title);
+        dialogBuilder.setMessage(R.string.wallet_lock_reset_wallet_message);
+        //Inverting dialog answers to prevent accidental wallet reset
+        dialogBuilder.setNegativeButton(R.string.wallet_lock_reset_wallet_title, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                resetWallet();
+            }
+        });
+        dialogBuilder.setPositiveButton(android.R.string.no, null);
+        dialogBuilder.setCancelable(false);
+        Dialog dialog = dialogBuilder.create();
+        dialog.setCanceledOnTouchOutside(false);
+        dialog.show();
+    }
+
+    private void resetWallet() {
+        final WalletLock walletLock = WalletLock.getInstance();
+        if (WalletLock.getInstance().isWalletLocked(wallet)) {
+            UnlockWalletDialogFragment.show(getSupportFragmentManager(), new DialogInterface.OnDismissListener() {
+                @Override
+                public void onDismiss(DialogInterface dialog) {
+                    if (!walletLock.isWalletLocked(wallet)) {
+                        resetWallet();
+                    }
+                }
+            });
+            return;
+        }
+
+        log.info("Clearing all app data and exiting.");
+        try {
+            Runtime.getRuntime().exec("pm clear "+ getApplicationContext().getPackageName());
+        } catch (IOException e) {
```

Comment:
We already have a method `WalletApplication.eraseAndCreateNewWallet()` that handles removing the wallet.  This method does not remove any other files (such as the logs, which we want to keep in the event there is an issue with the wallet reset function).  Additionally, it will create a new empty wallet.

Another place to look that handles this is when the PIN is entered incorrectly to many times in is in `PinRetryController.showResetWalletDialog`

Reply:
In our first implementation of this reset functionality, all the logs were deleted as well as all other files.  Then when we got user reports of issues, there were no logs detailing what previously happened.

Reply:
Tried `WalletApplication.eraseAndCreateNewWallet()` - it doesn't start the process from scratch (OnboardingActivity). I believe as a user i want all the data to be removed, like app was reinstalled, don't you think so?

Reply:
For the user it feels like app has crashed when i do `WalletApplication.eraseAndCreateNewWallet()`

Reply:
With the redesign, the flow and other code will need to be updated or changed to make it go to the OnboardingActivity.  `WalletApplication.eraseAndCreateNewWallet()` only handles the wallet file and wallet object (deleting and creating a new file).  It doesn't do any UI.  The point of my first comment is to use `WalletApplication.eraseAndCreateNewWallet()` instead of 
`Runtime.getRuntime().exec("pm clear "+ getApplicationContext().getPackageName());` because the `pm clear` will delete other files that should remain.